seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/util/KuduUtil.java

```diff
@@ -0,0 +1,139 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.kudu.util;
+
+import org.apache.seatunnel.common.exception.CommonErrorCode;
+import org.apache.seatunnel.connectors.seatunnel.kudu.config.CommonConfig;
+import org.apache.seatunnel.connectors.seatunnel.kudu.config.KuduSourceConfig;
+import org.apache.seatunnel.connectors.seatunnel.kudu.exception.KuduConnectorErrorCode;
+import org.apache.seatunnel.connectors.seatunnel.kudu.exception.KuduConnectorException;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.authentication.util.KerberosName;
+import org.apache.kudu.Schema;
+import org.apache.kudu.client.AsyncKuduClient;
+import org.apache.kudu.client.KuduClient;
+import org.apache.kudu.client.KuduScanToken;
+import org.apache.kudu.client.KuduTable;
+
+import lombok.extern.slf4j.Slf4j;
+import sun.security.krb5.Config;
+import sun.security.krb5.KrbException;
+
+import java.io.IOException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+public class KuduUtil {
+
+    private static final String ERROR_MESSAGE =
+            "principal and keytab can not be null current principal %s keytab %s";
+
+    public static final String KRB5_CONF_KEY = "java.security.krb5.conf";
+
+    public static final String HADOOP_AUTH_KEY = "hadoop.security.authentication";
+
+    public static final String KRB = "kerberos";
+
+    public static KuduClient getKuduClient(CommonConfig config) {
+        try {
+            if (config.getKeytab() != null && config.getPrincipal() != null) {
+
+                UserGroupInformation ugi = loginAndReturnUgi(config);
+                return ugi.doAs(
+                        (PrivilegedExceptionAction<KuduClient>)
+                                () -> getKuduClientInternal(config));
+            }
+
+            return getKuduClientInternal(config);
+
+        } catch (IOException | InterruptedException e) {
+            throw new KuduConnectorException(KuduConnectorErrorCode.INIT_KUDU_CLIENT_FAILED, e);
+        }
+    }
+
+    private static UserGroupInformation loginAndReturnUgi(CommonConfig config) throws IOException {
+        if (StringUtils.isBlank(config.getPrincipal()) || StringUtils.isBlank(config.getKeytab())) {
+            throw new KuduConnectorException(
+                    CommonErrorCode.ILLEGAL_ARGUMENT,
+                    String.format(ERROR_MESSAGE, config.getPrincipal(), config.getKeytab()));
+        }
+        if (StringUtils.isNotBlank(config.getKrb5conf())) {
+            reloadKrb5conf(config.getKrb5conf());
+        }
+        Configuration conf = new Configuration();
+        conf.set(HADOOP_AUTH_KEY, KRB);
+        UserGroupInformation.setConfiguration(conf);
```

Comment:
`UserGroupInformation ` is s static Class, If there are more than one place use `UserGroupInformation.setConfiguration(conf);` in one JVM, this will generate mutual coverage. So new a UserGroupInformation Object is more suggested approach.